src/main/java/edu/ie3/io/processor/EntityProcessor.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.io.processor;
+
+import edu.ie3.exceptions.EntityProcessorException;
+import edu.ie3.exceptions.FactoryException;
+import edu.ie3.models.StandardUnits;
+import edu.ie3.models.UniqueEntity;
+import edu.ie3.models.result.ResultEntity;
+import edu.ie3.util.TimeTools;
+import edu.ie3.util.quantities.interfaces.HeatCapacity;
+import java.beans.Introspector;
+import java.lang.reflect.Method;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.*;
+import javax.measure.Quantity;
+import javax.measure.quantity.ElectricCurrent;
+import javax.measure.quantity.Power;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Internal API Interface for EntityProcessors. Main purpose is to 'de-serialize' models into a
+ * fieldName -> value representation to allow for an easy processing into a database or file sink
+ * e.g. .csv
+ *
+ * @version 0.1
+ * @since 31.01.20
+ */
+public abstract class EntityProcessor<T extends UniqueEntity> {
+
+  public final Logger log = LogManager.getLogger(this.getClass());
+  private final Class<? extends T> registeredClass;
+  protected final String[] headerElements;
+  protected final LinkedHashMap<String, Method> fieldNameToMethod = new LinkedHashMap<>();
+  protected final boolean resultModel;
+
+  /** Field name of {@link UniqueEntity} uuid */
+  private static final String uuidString = "uuid";
+
+  /**
+   * Create a new EntityProcessor
+   *
+   * @param registeredClass the class the entity processor should be able to handle
+   */
+  public EntityProcessor(Class<? extends T> registeredClass) {
+    this.registeredClass = registeredClass;
+    this.resultModel = isResultModel(registeredClass);
+    this.headerElements = registerClass(registeredClass);
+    TimeTools.initialize(ZoneId.of("UTC"), Locale.GERMANY, "yyyy-MM-dd HH:mm:ss");
+  }
+
+  /**
+   * Determines if the provided class is a subclass of ResultEntity
+   *
+   * @param cls the provided class
+   * @return true if it is a subclass of ResultEntity, false otherwise
+   */
+  private boolean isResultModel(Class<?> cls) {
+    return ResultEntity.class.isAssignableFrom(cls);
+  }
+
+  /**
+   * Register the class provided in the constructor
+   *
+   * @param cls class to be registered
+   * @return an array of strings of all field values of the registered class
+   */
+  private String[] registerClass(Class<?> cls) {
+
+    try {
+      Arrays.stream(Introspector.getBeanInfo(cls, Object.class).getPropertyDescriptors())
```

Comment:
Isn't this the devil's method?

Reply:
Yep, but we need it here to determine the fields we want to write. In contrast to the application where it causes performance issues, here it is only used once to register the class that should be written out. Hence, it is acceptable as it is a one shot call and not (like in the other application) called for each element to be processed.